### PR TITLE
[TISPARK-9]Optimize scan plan cost model

### PR DESCRIPTION
**Pull request by Novemser**
Adjust some CBO parameters
1. Refine double read cost.
2. Table size estimation.

### CODE REVIEW

File: tikv-client/src/main/java/com/pingcap/tikv/predicates/ScanAnalyzer.java [hunk 1]
```diff
@@ -153,6 +153,7 @@ public ScanPlan buildScan(List<TiColumnInfo> columnList, List<Expression> condit
       // table name, index and handle column
       int indexSize = index.getIndexColumns().size() + 2;
       if (isDoubleRead) {
+        cost *= 4; // we assume double read's cost is around 4 times of simple table scan.
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/predicates/ScanAnalyzer.java]: Could you plz explain why the factor is multiplied over the whole cost here?

From my perspective this factor should be EITHER multiplied over tableSize (indicating IO for each table row) OR indexSize (IO for each index row).

Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/predicates/ScanAnalyzer.java]: If a factor is applied to `indexSize`, the same factor should be applied to the `indexSize` in the else-clause

File: core/src/main/scala/com/pingcap/tispark/statistics/StatisticsManager.scala [hunk 2]
```diff
@@ -240,6 +240,61 @@ class StatisticsManager(tiSession: TiSession) {
...
+      case TypeDouble => 8
+      case TypeDecimal => 10
+      case TypeNewDecimal => 10
+      case TypeNull => 1
+      case TypeTimestamp => 4
+      case TypeDate => 3
+      case TypeYear => 1
+      case TypeDatetime => 8
+      case TypeDuration => 3
+      case TypeString => 255 * goldenSplitFactor
```
[the remark this answers is not in the document]
Comment (Novemser) [reply core/src/main/scala/com/pingcap/tispark/statistics/StatisticsManager.scala]: A higher level of abstraction is done in new commits.

**Comment by Novemser** [comment]
PTAL, thx @birdstorm

**Pull request merged**